src/MenuStyle.php

```diff
@@ -267,13 +267,14 @@ public function getWidth() : int
 
     public function setWidth(int $width) : self
     {
-        $availableWidth = $this->terminal->getWidth() - ($this->margin * 2) - ($this->padding * 2);
-
-        if ($width >= $availableWidth) {
-            $width = $availableWidth;
+        if ($width >= $this->terminal->getWidth()) {
+            $width = $this->terminal->getWidth();
         }
 
         $this->width = $width;
+        if ($this->margin === -1) {
```

Comment:
If we want to center the menu we force recalculation of margin by setting it again.